### PR TITLE
Add random race to matchup players.

### PR DESCRIPTION
**Pull request by aaronjewell**
Use the player's hero selection to determine, if possible,
which race they were assigned from the random selection.
Assigned to a new member property, so that it can be used
separately for statistical computations/display. If the
player's race is not random, then null is assigned to the
random race member. If it cannot be determined which race
was assigned, then random is set for both the player race
as well as the random race.

First step required for handling w3champions/w3champions-ui#281
sanely in multiple UI components. Also, computed in the record so
that statistics can be computed at a later time.

Does not backfill previous matches.

### CODE REVIEW

**Comment by aaronjewell** [comment]
C# is not my primary language, and I don't have ReSharper or similar helpers installed, so it's likely there is a more succinct way to write some of the race determination code. I would love any feedback! 😃

**Comment by abeninski** [comment]
Hey! Thanks for the PR!

Heroes are not the most reliable way to determine race.
We discovered that the game reports the race, so you can use that.

In the `MatchFinishedEvent` you can see there is a `result` property and in in `raceId`
raceId: 1 = human
raceId: 2 = orc
raceId: 3 = undead
raceId: 4 = elf

**Comment by aaronjewell** [comment]
Even better! Updated to use this result property.

File: W3ChampionsStatisticService/Matches/Matchup.cs [hunk 1]
```diff
@@ -68,6 +68,17 @@ public static Matchup Create(MatchFinishedEvent matchFinishedEvent)
                 .ThenBy(x => x.team)
                 .ToList();
 
+            foreach (var resultPlayer in matchFinishedEvent.result.players)
```
Comment (abeninski) [inline W3ChampionsStatisticService/Matches/Matchup.cs:71]: In some cases result is null. Can you just put 1 if and it will be fine :)

**Comment by aaronjewell** [comment]
@abeninski - Thanks for catching that! How do you feel about this update? Rather than default them to a specific race for the rndRace member property when there is no result in the event, we can set it as random. That way if we want to use this for stat calculation later, it doesn't skew the results towards a specific race.

**Comment by aaronjewell** [comment]
Added a test for the null result, and moved the Race enum extensions into its own file to follow the conventions of the other enums.

**Pull request merged**